aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -157,50 +157,28 @@ at::Tensor& random_from_to_impl(at::Tensor& self, int64_t from, c10::optional<in
 
 // ==================================================== Normal ========================================================
 
-// This function computes broadcasted size of mean and std, resize the output to the broadcasted size if it was empty
-// [Note] The following features will be deprecated in version 1.6 release and function signature will be changed after
-//   When mean and std are not broadcastable but have same number of elements:
-//     This function will resize the output to the size of mean if it was empty.
-//     This function will reshape the std to the shape of mean.
-//     This function will return true in deprecated case, false in broadcastable case and throw in all other cases before deprecation.
-//     This function will not return and throw if mean and std are not broadcastable after deprecation
-static bool resize_output_for_normal(at::Tensor& output, const at::Tensor& mean, const at::Tensor& std) {
-  bool expandable = at::are_expandable(mean.sizes(), std.sizes());
-  bool empty_output = output.numel() == 0;
-
-  if (expandable) {
-    auto shape = at::infer_size(mean.sizes(), std.sizes());
-    TORCH_CHECK(
-        empty_output || output.sizes().equals(shape),
-        "inconsistent tensor, output size (", output.sizes(), ") is not the same as broadcasted mean and std size (", shape, ")");
-    if (empty_output) {
-      at::native::resize_(output, shape);
-    }
-    return false;
-  }
-  else {
-    TORCH_CHECK(
-        mean.numel() == std.numel(),
-        "inconsistent tensor, std and mean are not broadcastable and have different number of elements, "
-        "expected mean ", mean.sizes(), " and std ", std.sizes(), " to have same number of elements)");
-    TORCH_CHECK(
-        empty_output || output.sizes().equals(mean.sizes()),
-        "inconsistent tensor, std and mean are not broadcastable, output size (", output.sizes(), ") is not the same as mean size (", mean.sizes(), ")");
-    TORCH_WARN_ONCE(
-        "std and mean have the same number of elements, but are not broadcastable. This was previously a "
-        "supported mode of operation, but is now deprecated and the support will be removed in version 1.6 release. "
-        "Note that the current implementation reshapes std to the shape of mean, which may be incur data copies. "
-        "Please ensure that std and mean are broadcastable to avoid these issues.");
-    if (empty_output) {
-      at::native::resize_(output, mean.sizes());
-    }
-    return true;
-  }
-}
+#define CHECK_NORMAL_SIZES(output, input) \
+  TORCH_CHECK( \
+    output.sizes().equals(input.sizes()), \
+    "inconsistent tensor, output size (", output.sizes(), \
+    ") is not the same as input size (", input.sizes(), ")");
+
```

Comment:
It looks like this check is not necessary now, as it should be performed by the structured kernels machinery.

Reply:
We should modify that code to call into the `at::normal` functions.

Reply:
> We should only avoid calling the at:: API when the performance is crucial

could you elaborate on the perf bit?

Reply:
PyTorch has a custom dispatcher mechanism. When you call a function in the `at::` namespace, this mechanism is activated and it calls the correct backend, e.g. CPU, CUDA, XLA, but also forward or backward AD. This has certain overhead, so sometimes you'll find code that calls directly to the corresponding `at::native::` function. Now, most of the time, the correct thing to do is to call the `at::` function, but if you are implementing your own backward AD (i.e. it's not a composite kernel) then you should be good to call `at::native::`. See a non-example of this where I did and shouldn't have called an `at::native::` within a forward AD implementation: https://github.com/pytorch/pytorch/pull/63568#discussion_r749699629